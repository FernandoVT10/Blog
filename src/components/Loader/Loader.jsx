import "./Loader.scss";

export default ({ active,  height = 300 }) => {
    const loaderClass = active ? "loader--active" : "";
    
    return (
        <div
        className={`loader ${loaderClass}`}
        style={{ height }}>
            <div
            className="spinner-border text-light loader__spinner"
            role="status">
                <span className="sr-only">Loading...</span>
            </div>
        </div>
    );
};