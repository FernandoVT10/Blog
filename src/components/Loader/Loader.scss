.loader {
    width: 100%;
    display: none;
    justify-content: center;
    align-items: center;

    &--active {
        display: flex;
    }

    &__spinner {
        width: 50px;
        height: 50px;
    }
}