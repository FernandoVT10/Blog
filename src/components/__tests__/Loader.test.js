import { create } from "react-test-renderer";
import Loader from "../Loader/";

test("Check if <Loader/> renders correctly", () => {
    const component = create(<Loader active={true} />);

    expect(component.toJSON()).toMatchSnapshot();
});