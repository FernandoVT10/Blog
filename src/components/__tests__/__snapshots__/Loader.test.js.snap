// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <Loader/> renders correctly 1`] = `
<div
  className="loader loader--active"
  style={
    Object {
      "height": 300,
    }
  }
>
  <div
    className="spinner-border text-light loader__spinner"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
